roles/haproxy/handlers/main.yml

```diff
@@ -1,17 +1,17 @@
 ---
 
 - name: Restart haproxy service
-  systemd:
+  ansible.builtin.systemd:
     daemon_reload: true
     name: haproxy
     enabled: true
     state: restarted
   listen: "restart haproxy"
 
 - name: Check HAProxy is started and accepting connections
-  wait_for:
+  ansible.builtin.wait_for:
     port: "{{ haproxy_listen_port.stats }}"
-    host: "{{ hostvars[inventory_hostname]['inventory_hostname'] }}"
+    host: "{{ cluster_vip | default(hostvars[inventory_hostname]['inventory_hostname'], true) }}"
```

Comment:
I think this is an incorrect check because only one haproxy will be checked the one that is available by VIP

the initial task of this check was to check the status of each haproxy for each server in the balancers group

Reply:
You can add a separate task that will check availability by VIP

Reply:
"Check if HAProxy is accepting connections via the VIP address"

Reply:
but this won't work because the keepalived role is executed later  

in this case, keepalived role must be performed before haproxy

Reply:
yes, you're right. But shouldn't the keepalived role be executed before haproxy anyways?

Reply:
I don't have to, but I can do it.

Reply:
Well, in the end this PR is just cosmetics. So it would be also fine to leave it as it is.

Reply:
Please close this PR if you do not plan to further refine this part of the code.